webapp/lib/model.g.dart

```diff
@@ -44,51 +44,63 @@ class Conversation {
     };
   }
 
-  Future<bool> addToTagId(DocPubSubUpdate pubSubClient, String valueToAdd) {
+  /// Add [valueToAdd] to tagIds in the receiver.
+  /// Callers should catch and handle IOException.
+  Future<void> addToTagId(DocPubSubUpdate pubSubClient, String valueToAdd) {
     return addToTagId_forAll(pubSubClient, [this], valueToAdd);
   }
 
-  static Future<bool> addToTagId_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, String valueToAdd) async {
+  /// Add [valueToAdd] to tagIds for each Conversation.
+  /// Callers should catch and handle IOException.
+  static Future<void> addToTagId_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, String valueToAdd) async {
     final docIdsToPublish = <String>[];
     for (var doc in docs) {
       if (!doc.tagIds.contains(valueToAdd)) {
         doc.tagIds.add(valueToAdd);
         docIdsToPublish.add(doc.docId);
       }
     }
-    if (docIdsToPublish.isEmpty) return true;
+    if (docIdsToPublish.isEmpty) return;
     return pubSubClient.publishDocAdd(collectionName, docIdsToPublish, {"tags": [valueToAdd]});
   }
 
-  Future<bool> setTagIds(DocPubSubUpdate pubSubClient, Set<String> newValue) {
+  /// Set tagIds for the receiver.
+  /// Callers should catch and handle IOException.
+  Future<void> setTagIds(DocPubSubUpdate pubSubClient, Set<String> newValue) {
     return setTagIds_forAll(pubSubClient, [this], newValue);
   }
 
-  static Future<bool> setTagIds_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, Set<String> newValue) async {
+  /// Set tagIds for each Conversation.
+  /// Callers should catch and handle IOException.
+  static Future<void> setTagIds_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, Set<String> newValue) async {
     final docIdsToPublish = <String>[];
     for (var doc in docs) {
       if (doc.tagIds != newValue) {
         doc.tagIds = newValue;
         docIdsToPublish.add(doc.docId);
       }
     }
-    if (docIdsToPublish.isEmpty) return true;
+    if (docIdsToPublish.isEmpty) return;
     return pubSubClient.publishDocChange(collectionName, docIdsToPublish, {"tags": newValue?.toList()});
   }
 
-  Future<bool> removeFromTagId(DocPubSubUpdate pubSubClient, String valueToRemove) {
+  /// Remove [valueToRemove] from tagIds in the receiver.
+  /// Callers should catch and handle IOException.
+  Future<void> removeFromTagId(DocPubSubUpdate pubSubClient, String valueToRemove) {
     return removeFromTagId_forAll(pubSubClient, [this], valueToRemove);
   }
 
-  static Future<bool> removeFromTagId_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, String valueToRemove) async {
+  /// Remove [valueToRemove] from tagIds in each Conversation.
+  /// Callers should catch and handle IOException.
+  static Future<void> removeFromTagId_forAll(DocPubSubUpdate pubSubClient, List<Conversation> docs, String valueToRemove) async {
```

Comment:
I think this should be named either `removeTagId` or `removeFromTagIds`? I have a slight preference for the former option, but your call.

Reply:
Good catch. I need to distinguish between removing a tagId from the list of tagIds in a single conversation and doing the same in multiple conversations, and I've been struggling what to name the latter. I've changed them to
* `addTagId` - add a single tag id to the tagIds in this Conversation
* `addTagIdToAll` - add a single tag id to the tagIds in each specified Conversation
* `setTagIds` - set the tagIds in this Conversation
* `setTagIdsForAll - set the tagIds for each specified Conversation
* `removeTagId` - remove a single tag id from the tagIds in this Conversation
* `removeTagIdFromAll` - remove a single tag id from the tagIds in each specified Conversation
Happy to discuss and rename again :-)